src/tribler/gui/start_gui.py

```diff
@@ -67,22 +72,25 @@ def run_gui(api_port, api_key, root_state_dir, parsed_args):
                 elif arg.startswith('magnet'):
                     app.send_message(arg)
             logger.info('Close the current application.')
-            sys.exit(1)
+            process_locker.sys_exit(1, 'Tribler GUI application is already running')
 
         logger.info('Start Tribler Window')
         window = TriblerWindow(app_manager, settings, root_state_dir, api_port=api_port, api_key=api_key)
         window.setWindowTitle("Tribler")
         app.tribler_window = window
         app.parse_sys_args(sys.argv)
-        sys.exit(app.exec_())
+        exit_code = app.exec_()
+        process_locker.sys_exit(exit_code or None)
 
     except ImportError as ie:
         logger.exception(ie)
-        error_and_exit("Import Error", f"Import error: {ie}")
+        show_system_popup("Import Error", f"Import error: {ie}")
+        process_locker.sys_exit(1, exc=ie)
 
     except TriblerException as te:
         logger.exception(te)
-        error_and_exit("Tribler Exception", f"{te}")
+        show_system_popup("Tribler Exception", f"{te.__class__.__name__}: {te}")
+        process_locker.sys_exit(1, exc=te)
```

Comment:
This block could be replaced by

```python
    except (ImportError, TriblerException) as e:
        logger.exception(e)
        exception_name = e.__class__.__name__
        show_system_popup(exception_name, f"{exception_name}: {e}")
        process_locker.sys_exit(1, exc=e)
```

But is still unclear why `ImportError` and `TriblerException` are special :)

Reply:
Done